jetty-runner/src/main/java/org/eclipse/jetty/runner/Runner.java

```diff
@@ -553,6 +555,30 @@ protected void initClassLoader()
 
     public static void main(String[] args)
     {
+        System.err.println("WARNING:");
+        System.err.println("         jetty-runner is now deprecated with no direct replacement.");
+        System.err.println("         We encourage you to migrate to jetty-home soon.");
+        System.err.println("IMPORTANT NOTE:");
+        System.err.println("         When public Java 8 updates stop, jetty-runner will be fully deprecated");
+        System.err.println("         and removed from the Jetty codebase.");
+        System.err.println("Oracle Java SE Support Roadmap:");
+        System.err.println("         http://www.oracle.com/technetwork/java/javase/eol-135779.html");
+
+        try
+        {
+            String jvmVersionStr = System.getProperty("java.vm.specification.version");
+            int jvmVersion = Integer.parseInt(jvmVersionStr);
+            if(jvmVersion >= 9)
+            {
+                System.err.println();
+                System.err.printf("ERROR: Java 9+ (your JVM is %s) does not support jetty-runner%n", jvmVersionStr);
```

Comment:
That's not true - the jetty-runner certainly works on jdk-9 (tested with openjdk build 9.0.4+11).

Reply:
depends on what the user uses.
simple webapps can limp along and work.
but the minute they step out of that cuddly spot they run smack into the JEP-238 issues (for one). #1905
and the ServiceLoader issues.
and the HTTPS issues.
and the HTTP/2 issues.
and the JSP/EL issues.

the "generic" uber-jar approach is a dead end.
enhancements to shade plugin will not save us.
perhaps an alternative approach is in our future, one that doesn't repackage the jars, but uses them from a parent archive? (wonder if nested zipfs is supported?)

Reply:
There's 2 different things here: there's the jetty-runner build and the webapp itself. For the jetty-runner, it will currently work on java-9 (with jsps, annotations etc) because we've built it to work on java-9 (and none of our dependencies so far have multi-release jars with multiple copies of classes in them).  As I said in #1905 that's a fragile situation because sooner or later that will change. So I'm not disputing the need to deprecate jetty-runner. I'm just saying that its not strictly true to say that it doesn't work on java-9: currently it does work, but at some future point it won't. We just need to tell folks that and point them to alternatives.